eventhubs/client/README.md

```diff
@@ -0,0 +1,347 @@
+when creating an event producer. 
+
+#### Partition ID
+
+```java
+ EventHubProducerOptions producerOptions = new EventHubProducerOptions().partitionId(partitionId);
+ EventHubProducer producer = client.createProducer(producerOptions);
+```
+Many Event Hub operations take place within the scope of a specific partition. Because partitions are owned by the 
+Event Hub, their names are assigned at the time of creation. To understand what partitions are available, You can use 
+the `getPartitionIds` function to get the ids of all available partitions in your Event Hub instance.
+
+```java
+private static final Duration OPERATION_TIMEOUT = Duration.ofSeconds(30);
+...
+
+// We take the first partition id.
+// .blockFirst() here is used to synchronously block until the first partition id is emitted. The maximum wait
+// time is set by passing in the OPERATION_TIMEOUT value. If no item is emitted before the timeout elapses, a
+// TimeoutException is thrown.
+
+String firstPartition = client.getPartitionIds().blockFirst(OPERATION_TIMEOUT);
+```
+#### Partition Key
+
+When an Event Hub producer is not associated with any specific partition, it may be desirable to request that the Event
+ Hubs service keep different events or batches of events together on the same partition. This can be accomplished by 
+ setting a `partition key` when publishing the events.
+ 
+```java
+// The partition key is NOT the identifier of a specific partition. Rather, it is an arbitrary piece of string data
+// that Event Hubs uses as the basis to compute a hash value. Event Hubs will associate the hash value with a specific
+// partition, ensuring that any events published with the same partition key are rerouted to the same partition.
+//
+// Note that there is no means of accurately predicting which partition will be associated with a given partition key;
+// we can only be assured that it will be a consistent choice of partition. If you have a need to understand which
+// exact partition an event is published to, you will need to use an Event Hub producer associated with that partition.
+
+SendOptions sendOptions = new SendOptions().partitionKey("ANY_PARTITION_KEY");
+producer.send(dataList, sendOptions).subscribe(
+    ...
+);
+```
+
+### Consume events from an Event Hub
+
+In order to consume events, you'll need to create an EventHubConsumer for a specific partition and consumer group 
+combination. When an Event Hub is created, it starts with a default consumer group that can be used to get started. 
+A consumer also needs to specify where in the event stream to begin receiving events; in our example, we will focus 
+on reading new events as they are published.
+
+#### Consumer creation
+
+```java
+EventHubConsumer consumer = client.createConsumer(EventHubClient.DEFAULT_CONSUMER_GROUP_NAME, partitionID, 
+    EventPosition.latest());
+```
+
+#### Consume events
+
+```java
+private static final int NUMBER_OF_EVENTS = 10;
```

Comment:
consume events can be something as simple as:

```
consumer.receive().subscribe(event -> {
   // do stuff.
});
````

you don't need to show everything